### DISCUSSION TITLE
Can I enable FocusTrap conditionally (using FocusTrap.features.None, etc)?

### DISCUSSION THREAD

**Discussion by ragulka** [tailwindlabs/headlessui#2416] [Help]
I am trying to enable FocusTrap only when a certain condition is true. I see that FocusTrap has a features prop, to which I can pass something like `FocusTrap.features.None`, or `FocusTrap.features.All` or `FocusTrap.features.TabLock | FocusTrap.features.FocusLock`, etc.

I would imagine if I do the following, I should get what I need:

```
<FocusTrap :features="enableFocusTrap ? FocusTrap.features.All : FocusTrap.features.None">
   // content here...
</FocusTrap>
```

However, it looks like `FocusTrap.features.None` still prevents the user from tabbing outside the FocusTrap component. This seems to happen because [`handleBlur` here](https://github.com/tailwindlabs/headlessui/blob/2063132f1a7410c4491fc7aec60525233c5f95ea/packages/%40headlessui-vue/src/components/focus-trap/focus-trap.ts#L139-L173) does not account for the feature flags.

I understand that `FocusTrap` is an undocumented feature/component, so perhaps I am missing something else as well.

The main question is - shouldn't the `FocusTrap.features.None` flag disable _all_ focus-trapping functionality?

**Comment by hobadams** [comment]
I'd love to use this too. I'd like to wrap a sidebar in this component if possible

**Comment by spaceemotion** [comment]
I too would like to disable focus trapping in modals - My application already uses the `focus-trap` package and they're getting confused. It would be much easier if I can share the focus stack from `focus-trap` across all components, and not have HeadlessUI handle its own trapping.
